internal/manifest/v1alpha1/spec_resolver.go

```diff
@@ -0,0 +1,316 @@
+			}
+		}
+	}
+
+	// filter config for install
+	chartConfig, chartValues, err := parseChartConfigAndValues(configs, name)
+	if err != nil {
+		return nil, err
+	}
+	values := make(map[string]any, len(chartConfig)+len(chartValues))
+	for k, v := range chartConfig {
+		values[k] = v
+	}
+	for k, v := range chartValues {
+		values[k] = v
+	}
+
+	return values, nil
+}
+
+func parseInstallConfigs(decodedConfig interface{}) ([]interface{}, error) {
+	var configs []interface{}
+	installConfigObj, decodeOk := decodedConfig.(map[string]interface{})
+	if !decodeOk {
+		return nil, fmt.Errorf("reading install %s resulted in an error for "+v1alpha1.ManifestKind, ".spec.config")
+	}
+	if installConfigObj["configs"] != nil {
+		var configOk bool
+		configs, configOk = installConfigObj["configs"].([]interface{})
+		if !configOk {
+			return nil, fmt.Errorf(
+				"reading install %s resulted in an error for "+v1alpha1.ManifestKind,
+				"chart config object of .spec.config",
+			)
+		}
+	}
+	return configs, nil
+}
+
+func (m *ManifestSpecResolver) getChartInfoForInstall(
+	ctx context.Context,
+	install v1alpha1.InstallInfo,
+	specType types.RefTypeMetadata,
+	keyChain authn.Keychain,
+) (*types.ChartInfo, error) {
+	var err error
+	switch specType {
+	case types.HelmChartType:
+		var helmChartSpec types.HelmChartSpec
+		if err = m.Codec.Decode(install.Source.Raw, &helmChartSpec, specType); err != nil {
+			return nil, err
+		}
+
+		return &types.ChartInfo{
+			ChartName: helmChartSpec.ChartName,
+			RepoName:  install.Name,
+			URL:       helmChartSpec.URL,
+		}, nil
+	case types.OciRefType:
+		var imageSpec types.ImageSpec
+		if err = m.Codec.Decode(install.Source.Raw, &imageSpec, specType); err != nil {
+			return nil, err
+		}
+
+		// extract helm chart from layer digest
+		chartPath, err := internal.GetPathFromExtractedTarGz(ctx, imageSpec, m.Insecure, keyChain)
+		if err != nil {
+			return nil, err
+		}
+
+		return &types.ChartInfo{
+			ChartName: install.Name,
+			ChartPath: chartPath,
+		}, nil
+	case types.KustomizeType:
+		var kustomizeSpec types.KustomizeSpec
+		if err = m.Codec.Decode(install.Source.Raw, &kustomizeSpec, specType); err != nil {
+			return nil, err
+		}
+
+		return &types.ChartInfo{
+			ChartName: install.Name,
+			ChartPath: kustomizeSpec.Path,
+			URL:       kustomizeSpec.URL,
+		}, nil
+	case types.NilRefType:
+		return nil, fmt.Errorf("empty image type")
+	}
+
+	return nil, fmt.Errorf(
+		"unsupported type %s of install", specType,
+	)
+}
+
+func parseChartConfigAndValues(
+	configs []interface{}, name string,
+) (map[string]interface{}, map[string]interface{}, error) {
+	configString, valuesString, err := getConfigAndValuesForInstall(configs, name)
+	if err != nil {
+		return nil, nil, fmt.Errorf("manifest encountered an error while parsing chart config: %w", err)
+	}
+
+	config := map[string]interface{}{}
+	if err := strvals.ParseInto(configString, config); err != nil {
+		return nil, nil, err
+	}
```

Comment:
this is not required as it is not taken over in declarative reconciler? We can still keep values, but helm flags (config) can be removed

Reply:
these config values are used to apply them to the values meaning that overrides will be applied as normally. So declarative will make use of them. Of course we can still think of removing the config layer if that is relevant.